### PR TITLE
feat(healthchecks): add clickhouse and redis health

### PR DESCRIPTION
**Pull request by hazzadous**
## Changes

1. add clickhouse to healthcheck endpoint
2. add celery broker to healthcheck endpoint
3. add cache to healthcheck endpoint
4. add healthcheck endpoints as a middleware such that we can short circuit to before middlewares that make requests to dependencies
5. Add some basic debug logging

I could have added these as separate PRs, which may have been better. I'm happy to break up as the PR is quite large.

## How did you test this code?

Unittests, some very rough playing around locally.

### CODE REVIEW

**Comment by hazzadous** [comment]
argh there are some test failures. I think CI services not quite the same as I'm running locally, will check.

File: posthog/health.py [hunk 1]
```diff
@@ -112,6 +149,7 @@ def is_postgres_connected() -> bool:
         with connections[DEFAULT_DB_ALIAS].cursor() as cursor:
             cursor.execute("SELECT 1")
     except DjangoDatabaseError:
+        logger.debug("postgres_migrations_check_failure", exc_info=True)
```
Comment (guidoiaquinti) [inline posthog/health.py:152]: I think this is a bad copy paste, we should have a different logging here

File: posthog/health.py [hunk 2]
```diff
@@ -128,5 +166,88 @@ def are_postgres_migrations_uptodate() -> bool:
         executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
         plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
     except DjangoDatabaseError:
+        logger.debug("postgres_migrations_check_failure", exc_info=True)
         return False
+
     return False if plan else True
```
Comment (guidoiaquinti) [inline posthog/health.py:172]: this could be `bool(plan)`?

File: posthog/health.py [hunk 3]
```diff
@@ -64,30 +89,42 @@ def readyz(request):
...
+        "clickhouse": is_clickhouse_connected,
+        "postgres": is_postgres_connected,
+        "postgres_migrations_uptodate": are_postgres_migrations_uptodate,
+        "kafka_connected": is_kafka_connected,
+        "celery_broker": is_celery_broker_connected,
+        "cache": is_cache_backend_connected,
     }
 
     if role:
+        if role not in get_args(ServiceRole):
```
Comment (guidoiaquinti) [inline posthog/health.py:110]: I find it more readable like it was before (despite you've now optimized an `if`)

**Comment by hazzadous** [comment]
Tests finally passing 😠

**Review by guidoiaquinti — APPROVED** [review]
LGTM

**Pull request merged**